doc/LuaObject.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<class name="LuaObject" inherits="RefCounted" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/godotengine/godot/master/doc/class.xsd">
+	<brief_description>
+		Base class for all Lua objects in GDScript, except LuaState.
+	</brief_description>
+	<description>
+		The LuaObject class provides methods to access the associated Lua state and retrieve object pointers.
+		This class serves as the foundation for interacting with Lua objects from GDScript. It allows you to:
+		- Get the Lua state associated with the object
+		- Retrieve the pointer value of the Lua object
```

Comment:
Not sure if it's really necessary to talk about the internal object's pointer value here 🤔 If you think it's relevant, feel free to keep it, though.